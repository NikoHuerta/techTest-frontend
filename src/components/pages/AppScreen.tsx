import { AppLayout } from '../layout';
import { useAppSelector } from '../../hooks';
import { HomeScreen, ViewTaskScreen, NewTaskScreen, EditTaskScreen } from './';


export const AppScreen = () => {

  const page = useAppSelector(state => state.navigation.status);

  return (
    <>
        <AppLayout>
            { 
                page === 'allTasks' ? <ViewTaskScreen />
                : page === 'newTask' ? <NewTaskScreen />
                : page === 'editTask' ? <EditTaskScreen />
                : <HomeScreen />
            }
        </AppLayout>
    </>
  )
}
