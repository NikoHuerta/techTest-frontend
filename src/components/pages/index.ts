export * from './HomeScreen';
export * from './AppScreen';
export * from './NewTaskScreen';
export * from './ViewTaskScreen';
export * from './EditTaskScreen';