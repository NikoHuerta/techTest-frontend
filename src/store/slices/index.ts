export { default as taskReducer } from './taskSlice';
export { default as navigationReducer } from './navigationSlice';
export * from './navigationSlice';
export * from './taskSlice';