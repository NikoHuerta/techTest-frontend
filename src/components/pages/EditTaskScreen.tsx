import { ChangeEvent, FormEvent, useState } from 'react'; 

import { Box, Button, FormControlLabel, Grid, TextField, Checkbox } from '@mui/material';
import { useSnackbar } from 'notistack';
import moment from 'moment';
import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
import { AdapterMoment } from '@mui/x-date-pickers/AdapterMoment';
import { DateTimePicker } from '@mui/x-date-pickers/DateTimePicker';

import { useAppDispatch, useAppSelector } from '../../hooks';
import { setNavigationStatus, updateTask } from '../../store/slices';
import { ITask } from '../../interfaces';

export const EditTaskScreen = () => {

  const { enqueueSnackbar } = useSnackbar();
  const dispatch = useAppDispatch();
  const { objectId, descripcion, fechaCreacion, vigente } = useAppSelector(state => state.task.activeTask!);


  const [valueDescription, setValueDescription] = useState<string>(descripcion);
  const [touched, setTouched] = useState(false);
  const [valueDate, setValueDate] = useState<Date | null>( new Date(fechaCreacion) );
  const [valueVigente, setVigente] = useState<boolean | null>(vigente);


  const handleDateChange = (newValue: Date | null) => {
    setValueDate(newValue);
  };

  const handleVigenteChange = (event: ChangeEvent<HTMLInputElement>) => {
    setVigente(event.target.checked);
  };

  const handleDescriptionChange = (event: ChangeEvent<HTMLInputElement>) => {
    setValueDescription(event.target.value);
  };

  const handleUpdate = async (e: FormEvent<HTMLFormElement>) =>{
    e.preventDefault();
    if(valueDescription ==='') {
      return enqueueSnackbar('La descripcion no puede ser vacia', { 
        variant: 'error',
        autoHideDuration: 2000,
        anchorOrigin: {
          vertical: 'top',
          horizontal: 'right',
        } 
      });
    }
    
    const updatedTask: ITask = {
      descripcion : (valueDescription) ? valueDescription : '',
      fechaCreacion: moment(valueDate).format('YYYY-MM-DD HH:mm:ss'),
      vigente : (valueVigente) ? valueVigente : false,
      objectId,
    }

    try {
      const resp = await dispatch(updateTask(updatedTask));

      if(resp.meta.requestStatus === 'fulfilled') {
        enqueueSnackbar('Tarea editada correctamente', { 
          variant: 'success',
          autoHideDuration: 2000,
          anchorOrigin: {
            vertical: 'top',
            horizontal: 'right',
          } 
        });
      } else {
        enqueueSnackbar('Error en la comunicacion con backend', { 
          variant: 'error',
          autoHideDuration: 2000,
          anchorOrigin: {
            vertical: 'top',
            horizontal: 'right',
          } 
        });
      }

    } catch (e:any) {
      console.log('error', e);
    }
  }


  return (
    <>
      <h1>Edit Task Screen</h1>


      <form onSubmit={ handleUpdate } noValidate >
          
          <Grid container spacing={2} sx={{ mt:5 }}>
            <Grid item xs={6} sm={4}>
              <TextField
                label='Descripcion'
                name='descripcion'
                variant='filled'
                fullWidth
                required

                helperText={ valueDescription!.length === 0 && touched && 'Ingrese un valor'}
                error={ valueDescription!.length === 0 && touched }
                value={ valueDescription }
                onChange={ handleDescriptionChange }
                onBlur={ () => setTouched(true) }


              />
            </Grid>
            <Grid item xs={6} sm={4}>
              <LocalizationProvider dateAdapter={ AdapterMoment }>
                <DateTimePicker
                  label="Fecha de Creacion"
                  value={ valueDate }
                  onChange={ handleDateChange }
                  renderInput={( params ) => <TextField fullWidth variant='filled' name='fechaCreacion' {...params} />}
                />
              </LocalizationProvider>
              
            </Grid>  
            <Grid item xs={6} sm={4}>
              <FormControlLabel
                control={ 
                  <Checkbox checked={ valueVigente! } onChange={ handleVigenteChange } name='vigente' />
                }
                label='Vigente'
              />
            </Grid>    
          </Grid>

          <Box sx={{ mt: 5 }} display='flex' justifyContent='center'>
            
            <Button 
                type='button'
                color='primary' 
                className='circular-btn' 
                size='large'
                onClick={ () => { dispatch(setNavigationStatus({ status: 'allTasks' })) } }
              >
                Volver
            </Button>

            <Button 
              type='submit'
              color='primary' 
              className='circular-btn' 
              size='large'
            >
              Actualizar Tarea
            </Button>
          </Box>
      
      </form>

    </>
  )
}
