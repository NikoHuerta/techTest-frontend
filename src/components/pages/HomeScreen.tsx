import { Typography } from "@mui/material";
import { useEffect } from "react";
import { useAppDispatch } from "../../hooks";
import { loadTasks } from "../../store/slices";

export const HomeScreen = () => {

  const dispatch = useAppDispatch();

  useEffect(() => {
    dispatch(loadTasks());
  }, [dispatch]);
  
  
  return (
    <>
      <h1>Home Screen</h1>

      <Typography variant='subtitle1' gutterBottom sx={{ my: 5 }}>
        App encargada de manejar tareas. 
      </Typography>

      <Typography variant='subtitle2' gutterBottom sx={{ mb: 2 }}>
        Para ver las tareas, haga click en el boton "Ver Tareas".
      </Typography>

      <Typography variant='subtitle2' gutterBottom>
        Para agregar una nueva tarea, haga click en el boton "Nueva Tarea".
      </Typography>
      <Typography variant='subtitle2' gutterBottom>
        En la visualizacion usted podra Editar y Borrar una tarea respectiva.
      </Typography>
    </>
  )
}