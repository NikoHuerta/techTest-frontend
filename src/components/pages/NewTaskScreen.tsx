import { ChangeEvent, FormEvent, useState } from 'react'; 

import { Box, Button, FormControlLabel, Grid, TextField, Checkbox } from '@mui/material';
import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
import { AdapterMoment } from '@mui/x-date-pickers/AdapterMoment';
import { DateTimePicker } from '@mui/x-date-pickers/DateTimePicker';
import { useSnackbar } from 'notistack';
import moment from 'moment';

import { useAppDispatch } from '../../hooks';
import { addTask } from '../../store/slices';
import { ITask } from '../../interfaces/task';



export const NewTaskScreen = () => {

  const { enqueueSnackbar } = useSnackbar();
  const dispatch = useAppDispatch();

  const [valueDescription, setValueDescription] = useState<string | null>('');
  const [touched, setTouched] = useState(false);
  const [valueDate, setValueDate] = useState<Date | null>( new Date(moment.now()) );
  const [valueVigente, setVigente] = useState<boolean | null>(true);

  const handleDateChange = (newValue: Date | null) => {
    setValueDate(newValue);
  };

  const handleVigenteChange = (event: ChangeEvent<HTMLInputElement>) => {
    setVigente(event.target.checked);
  };

  const handleDescriptionChange = (event: ChangeEvent<HTMLInputElement>) => {
    setValueDescription(event.target.value);
  };

  const handleSubmit = async (e: FormEvent<HTMLFormElement>) =>{
    e.preventDefault();
    if(valueDescription ==='') {
      return enqueueSnackbar('Debe ingresar una descripcion', { 
        variant: 'error',
        autoHideDuration: 2000,
        anchorOrigin: {
          vertical: 'top',
          horizontal: 'right',
        } 
      });
    } 

    const newTask: ITask = {
      descripcion : (valueDescription) ? valueDescription : '',
      fechaCreacion: moment(valueDate).format('YYYY-MM-DD HH:mm:ss'),
      vigente : (valueVigente) ? valueVigente : false,
    }

    try {
      const resp = await dispatch(addTask(newTask));

      if(resp.meta.requestStatus === 'fulfilled') {
        enqueueSnackbar('Tarea agregada correctamente', { 
          variant: 'success',
          autoHideDuration: 2000,
          anchorOrigin: {
            vertical: 'top',
            horizontal: 'right',
          } 
        });
      } else {
        enqueueSnackbar('Error en la comunicacion con backend', { 
          variant: 'error',
          autoHideDuration: 2000,
          anchorOrigin: {
            vertical: 'top',
            horizontal: 'right',
          } 
        });
      }

    } catch (e:any) {
      console.log('error', e);
    }
      

  }


  return (
    <>
        <h1>New Task Screen</h1>
        
        <form onSubmit={ handleSubmit } noValidate >
          <Grid container spacing={2} sx={{ mt:5 }}>
            <Grid item xs={6} sm={4}>
              <TextField
                label='Descripcion'
                name='descripcion'
                variant='filled'
                fullWidth
                required

                helperText={ valueDescription!.length === 0 && touched && 'Ingrese un valor'}
                error={ valueDescription!.length === 0 && touched }
                value={ valueDescription }
                onChange={ handleDescriptionChange }
                onBlur={ () => setTouched(true) }


              />
            </Grid>
            <Grid item xs={6} sm={4}>
              <LocalizationProvider dateAdapter={ AdapterMoment }>
                <DateTimePicker
                  label="Fecha de Creacion"
                  value={ valueDate }
                  onChange={ handleDateChange }
                  renderInput={( params ) => <TextField fullWidth variant='filled' name='fechaCreacion' {...params} />}
                />
              </LocalizationProvider>
              
            </Grid>  
            <Grid item xs={6} sm={4}>
              <FormControlLabel
                control={ 
                  <Checkbox checked={ valueVigente! } onChange={ handleVigenteChange } name='vigente' />
                }
                label='Vigente'
              />
            </Grid>    
          </Grid>

          <Box sx={{ mt: 5 }} display='flex' justifyContent='center'>
            <Button 
              type='submit'
              color='primary' 
              className='circular-btn' 
              size='large'
            >
              Crear Tarea
            </Button>
          </Box>


        </form>
    </>
  )
}
