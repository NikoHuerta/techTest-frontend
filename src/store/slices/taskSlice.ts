import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
import { RootState } from '../store';

import { ITask } from '../../interfaces';
import { fetchAxios } from '../../helpers';
import { AxiosResponse } from 'axios';

export interface TaskState {
    tasks: ITask[];
    activeTask: ITask | null;
    status: 'idle' | 'loading' | 'failed';
}

const initialState: TaskState = {
    tasks: [],
    status: 'idle',
    activeTask: null,
};

//THUNK ASYNC ACTIONS
export const loadTasks = createAsyncThunk(
    'task/loadTasks',
    async () => {
        const response = await fetchAxios('task', undefined, 'GET') as AxiosResponse;
        const { data: body } = response;
        return body.tasks;
    }
);

export const addTask = createAsyncThunk(
    'task/addTask',
    async (task: ITask) => {
        const response = await fetchAxios('task', task , 'POST') as AxiosResponse;
        const { data: body } = response;
        return body.task;
    }
);

export const removeTask = createAsyncThunk(
    'task/removeTask',
    async (id: string) => {
        const response = await fetchAxios(`task/${ id }`, undefined, 'DELETE') as AxiosResponse;
        const { data: body } = response;
        return body.task;
    }
);

export const updateTask = createAsyncThunk(
    'task/updateTask',
    async ( task: ITask ) => {
        const response = await fetchAxios(`task/${ task.objectId }`, task, 'PUT') as AxiosResponse;
        const { data: body } = response;
        return body.task;
    }
);


export const taskSlice = createSlice({
    name: 'task',
    initialState,
    reducers: {
        
        setActiveTask: (state, action: PayloadAction<string>) => {
            state.activeTask = state.tasks.filter(task => task.objectId === action.payload)[0];
        },
        removeActiveTask: (state) => {
            state.activeTask = null;
        }
    },
    extraReducers: {

        //LOAD TASKS
        [loadTasks.pending.type]: (state) => {
            state.status = 'loading';
        },
        [loadTasks.fulfilled.type]: (state, action) => {
            state.status = 'idle';
            state.tasks = action.payload;
        },
        [loadTasks.rejected.type]: (state) => {
            state.status = 'failed';
        },

        //ADD TASK
        [addTask.pending.type]: (state) => {
            state.status = 'loading';
        },
        [addTask.fulfilled.type]: (state, action) => {
            state.tasks.push(action.payload);
            state.status = 'idle';
        },
        [addTask.rejected.type]: (state) => {
            state.status = 'failed';
        },

        //REMOVE TASK
        [removeTask.pending.type]: (state) => {
            state.status = 'loading';
        },
        [removeTask.fulfilled.type]: (state, action) => {
            state.tasks = state.tasks.filter(task => task.objectId !== action.payload.objectId);
            state.status = 'idle';
        },
        [removeTask.rejected.type]: (state) => {
            state.status = 'failed';
        },

        //UPDATE TASK
        [updateTask.pending.type]: (state) => {
            state.status = 'loading';
        },
        [updateTask.fulfilled.type]: (state, action) => {
            state.tasks = state.tasks.map(task => task.objectId === action.payload.objectId ? action.payload : task);
            state.status = 'idle';
        },
        [updateTask.rejected.type]: (state) => {
            state.status = 'failed';
        },
    }
        

});

// REDUCERS
export const { setActiveTask, removeActiveTask } = taskSlice.actions;

// SELECTORS
export const selectTasks = (state: RootState) => state.task.tasks;
export const selectActiveTask = (state: RootState) => state.task.activeTask;


export default taskSlice.reducer;
        



