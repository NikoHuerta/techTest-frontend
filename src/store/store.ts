import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import { taskReducer, navigationReducer } from './slices';

export const store = configureStore({
  reducer: {
    task: taskReducer,
    navigation: navigationReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
