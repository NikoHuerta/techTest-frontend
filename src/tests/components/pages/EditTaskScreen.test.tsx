import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import { Provider } from 'react-redux';
import { ThemeProvider, CssBaseline } from '@mui/material';
import { SnackbarProvider } from 'notistack';

import { darkTheme } from '../../../themes';
import { store } from '../../../store';
import { EditTaskScreen } from '../../../components/pages';
import { setActiveTask } from '../../../store/slices';
import { loadTasks } from '../../../store/slices/taskSlice';


describe('Pruebas en EditTaskScreen.tsx', () => {
    
    beforeAll( async () => {
        await store.dispatch(loadTasks());
        store.dispatch(setActiveTask('62b5f7f6bd7fc31ea94414f8'));
        store.getState().task.activeTask;
    });

    test('Debe de renderizarse correctamente', () => {

        const { asFragment } = render( 
            <Provider store={ store }>
                <ThemeProvider theme={ darkTheme }>
                    <SnackbarProvider>
    
                        <EditTaskScreen />
    
                    <CssBaseline />
                    </SnackbarProvider>
                </ThemeProvider>
            </Provider>);

        expect(asFragment()).toMatchSnapshot();
    });

    test('Debe de mostrar el titulo', () => {

        const { container } = render( 
            <Provider store={ store }>
                <ThemeProvider theme={ darkTheme }>
                    <SnackbarProvider>
    
                        <EditTaskScreen />
    
                    <CssBaseline />
                    </SnackbarProvider>
                </ThemeProvider>
            </Provider>);

        expect(container.querySelectorAll('h1')).toBeDefined();
    });

    test('Debe de mostrar el formulario de edicion', () => {

        const { container } = render( 
            <Provider store={ store }>
                <ThemeProvider theme={ darkTheme }>
                    <SnackbarProvider>
    
                        <EditTaskScreen />
    
                    <CssBaseline />
                    </SnackbarProvider>
                </ThemeProvider>
            </Provider>);
            
        expect(container.querySelectorAll('form')).toBeDefined();
    });



});