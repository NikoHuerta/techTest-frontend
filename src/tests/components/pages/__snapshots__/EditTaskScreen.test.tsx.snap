// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en EditTaskScreen.tsx Debe de renderizarse correctamente 1`] = `
<DocumentFragment>
  <h1>
    Edit Task Screen
  </h1>
  <form
    novalidate=""
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-hvca0x-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 css-1xd74lj-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required css-hum04-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for=":r0:"
            id=":r0:-label"
          >
            Descripcion
            <span
              aria-hidden="true"
              class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-125iqkp-MuiFormLabel-asterisk"
            >
              â€‰*
            </span>
          </label>
          <div
            class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-43673x-MuiInputBase-root-MuiFilledInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiFilledInput-input MuiInputBase-input css-1rv476z-MuiInputBase-input-MuiFilledInput-input"
              id=":r0:"
              name="descripcion"
              required=""
              type="text"
              value="descripcion 1"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 css-1xd74lj-MuiGrid-root"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-hum04-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for=":r1:"
            id=":r1:-label"
          >
            Fecha de Creacion
          </label>
          <div
            class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-43673x-MuiInputBase-root-MuiFilledInput-root"
          >
            <input
              aria-invalid="false"
              aria-label="Choose date, selected date is Jun 24, 2022"
              aria-readonly="true"
              class="MuiFilledInput-input MuiInputBase-input css-1rv476z-MuiInputBase-input-MuiFilledInput-input"
              id=":r1:"
              name="fechaCreacion"
              readonly=""
              type="text"
              value="06/24/2022 03:59 PM"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-4 css-1xd74lj-MuiGrid-root"
      >
        <label
          class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-jzcd7z-MuiFormControlLabel-root"
        >
          <span
            class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-1e6ud36-MuiButtonBase-root-MuiCheckbox-root"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input css-1m9pwf3"
              data-indeterminate="false"
              name="vigente"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CheckBoxIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
          >
            Vigente
          </span>
        </label>
      </div>
    </div>
    <div
      class="MuiBox-root css-144iql5"
    >
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeLarge MuiButton-textSizeLarge MuiButtonBase-root circular-btn css-f5p02u-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Volver
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeLarge MuiButton-textSizeLarge MuiButtonBase-root circular-btn css-f5p02u-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="submit"
      >
        Actualizar Tarea
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </form>
</DocumentFragment>
`;
